### PR TITLE
fix more clippy findings

### PR DESCRIPTION
**Pull request by matthiaskrgr**
* reduce references on match patterns (clippy::match_ref_pats)
* Use writeln!(fmt, "word") instead of write!(fmt, "word\n") (clippy::write_with_newline)
* libtest: remove redundant argument to writeln!() (clippy::writeln_empty_string)
* remove unneeded mutable references (cippy::unnecessary_mut_passed)
* libtest: declare variables as floats instead of casting them (clippy::unnecessary_cast)
* rustdoc: remove redundant static lifetimes (clippy::redundant_static_lifetimes)
* call .as_deref() instead of .as_ref().map(Deref::deref) (clippy::option_as_ref_deref)
* iterate over a maps values directly. (clippy::for_kv_map)
* rustdoc: simplify boolean condition (clippy::nonminimal_bool)
* Use ?-operator in more places (clippy::question_mark, had some false negatives fixed recently)
* rustdoc: Use .any(p) instead of find(p).is_some(). (clippy::search_is_some)
* rustdoc: don't call into_iter() on iterator. (clippy::identity_conversion)

### CODE REVIEW

File: src/libstd/sys_common/wtf8.rs [hunk 1]
```diff
@@ -615,8 +615,8 @@ impl Wtf8 {
         if len < 3 {
             return None;
         }
-        match &self.bytes[..3] {
-            &[0xED, b2 @ 0xB0..=0xBF, b3] => Some(decode_surrogate(b2, b3)),
+        match self.bytes[..3] {
```
Comment (lzutao) [inline src/libstd/sys_common/wtf8.rs]: The subslice pattern could be used here.

File: src/librustdoc/html/render.rs [hunk 2]
```diff
@@ -1542,7 +1542,7 @@ impl Context {
         }
 
         if self.shared.sort_modules_alphabetically {
-            for (_, items) in &mut map {
+            for items in &mut map.values_mut() {
```
Comment (ollie27) [inline src/librustdoc/html/render.rs]:
```suggestion
            for items in map.values_mut() {
```

File: src/libstd/sys_common/wtf8.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/libstd/sys_common/wtf8.rs]: I'm not too familiar with subslice patterns yet so I'll pass.

File: src/librustdoc/html/render.rs (hunk 2 above)
Comment (matthiaskrgr) [reply src/librustdoc/html/render.rs]: thanks

File: src/libstd/sys_common/wtf8.rs (hunk 1 above)
Comment (lzutao) [reply src/libstd/sys_common/wtf8.rs]: Resolving it in #70046

**Pull request merged**